src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,345 @@
+		if (ts.isIdentifier(importClause) && importClause.text === "sap/ui/core/UIComponent" &&
+			ts.isImportDeclaration(importClause.parent)) {
+			varName = importClause.parent?.importClause?.name?.getText() ?? "";
+		}
+		return varName;
+	}, "");
+
+	if (classDesc && ts.isClassDeclaration(classDesc)) {
+		const analysisResult = findAsyncInterface({
+			classDefinition: classDesc, manifestContent, checker, uiComponentImportVar});
+
+		if (analysisResult) {
+			reportResults(analysisResult, reporter, classDesc);
+		}
+	}
+}
+
+function mergeResults(a: AsyncInterfaceFindType, b: AsyncInterfaceFindType): AsyncInterfaceFindType {
+	// null = parent property does not exist i.e. rootView
+	// undefined = async flag is missing
+	// true|false = async flag is explicitly set
+
+	const compareValues = (aProp: null | undefined | boolean,
+		bProp: null | undefined | boolean): null | undefined | boolean => {
+		let result = null;
+		if (aProp === undefined || bProp === undefined) {
+			result = undefined;
+		}
+		if (typeof aProp === "boolean" || typeof bProp === "boolean") {
+			// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+			result = aProp || bProp;
+		}
+
+		return result;
+	};
+
+	return {
+		hasManifestDefinition: a.hasManifestDefinition || b.hasManifestDefinition,
+		routingAsyncFlag: compareValues(a.routingAsyncFlag, b.routingAsyncFlag),
+		rootViewAsyncFlag: compareValues(a.rootViewAsyncFlag, b.rootViewAsyncFlag),
+		hasAsyncInterface: compareValues(a.hasAsyncInterface, b.hasAsyncInterface),
+	};
+}
+
+function findAsyncInterface({classDefinition, manifestContent, checker, uiComponentImportVar}: {
+	classDefinition: ts.ClassDeclaration;
+	manifestContent: string | undefined;
+	checker: ts.TypeChecker;
+	uiComponentImportVar: string;
+}): AsyncInterfaceFindType | undefined {
+	if (!ts.isClassDeclaration(classDefinition)) {
+		return;
+	}
+
+	const returnTypeTemplate = {
+		hasAsyncInterface: null,
+		routingAsyncFlag: null,
+		rootViewAsyncFlag: null,
+		hasManifestDefinition: false,
+	} as AsyncInterfaceFindType;
+
+	// Checks the interfaces and manifest
+	const curClassAnalysis = classDefinition.members.reduce((acc, member) => {
+		const checkResult = doPropsCheck(member as ts.PropertyDeclaration, manifestContent);
+		return mergeResults(acc, checkResult);
+	}, {...returnTypeTemplate});
+
+	const heritageAnalysis =
+		classDefinition?.heritageClauses?.flatMap((parentClasses: ts.HeritageClause) => {
+			return parentClasses.types.flatMap((parentClass) => {
+				const parentClassType = checker.getTypeAtLocation(parentClass);
+
+				return parentClassType.symbol?.declarations?.flatMap((declaration) => {
+					let result = {...returnTypeTemplate} as AsyncInterfaceFindType;
+					// Continue down the heritage chain to search for
+					// the async interface or manifest flags
+					if (ts.isClassDeclaration(declaration) &&
+						(!uiComponentImportVar || declaration?.name?.getText() !== uiComponentImportVar)) {
```

Comment:
Some comments would be helpful for future readers. At least I'm a bit lost here.

`uiComponentImportVar` holds the name of the import variable for `sap/ui/core/UIComponent´ in the Component.js file that is being analyzed? Why is this relevant when analyzing the super classes?

Reply:
Thanks! It's not needed anymore for the purpose it served. The main reason behind it was that it was used to not go deeper than UIComponent.